File: src/main/java/com/puhovin/intellijplugin/twm/action/ToolWindowManagerActionGroup.java
package com.puhovin.intellijplugin.twm.action;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

public class ToolWindowManagerActionGroup extends DefaultActionGroup {

    @Override
    public @NotNull ActionUpdateThread getActionUpdateThread() {
        return ActionUpdateThread.EDT;
    }

    @Override
    public void update(AnActionEvent e) {
        Project project = e.getProject();

        e.getPresentation().setEnabledAndVisible(project != null && !project.isDefault());
    }
}

File: src/main/kotlin/com/puhovin/intellijplugin/twm/action/ToolWindowManagerActionGroup.kt
package com.puhovin.intellijplugin.twm.action

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.project.Project

/**
 * Action group for managing tool window actions.
 */
class ToolWindowManagerActionGroup : DefaultActionGroup() {

    /**
     * Gets the action update thread.
     *
     * @return The action update thread to be used for this action
     */
    override fun getActionUpdateThread(): ActionUpdateThread {
        return ActionUpdateThread.EDT
    }

    /**
     * Updates the action visibility and enablement based on the project state.
     *
     * @param e The action event containing the project context
     */
    override fun update(e: AnActionEvent) {
        val project: Project? = e.project
        e.presentation.isEnabledAndVisible = project != null
    }
}
